packages/api/lambdas/process-s3-dead-letter-archive.js

```diff
@@ -3,12 +3,69 @@
 const log = require('@cumulus/common/log');
 
 const { s3 } = require('@cumulus/aws-client/services');
-const { getJsonS3Object } = require('@cumulus/aws-client/S3');
+const { getJsonS3Object, deleteS3Object } = require('@cumulus/aws-client/S3');
+const S3 = require('@cumulus/aws-client/S3');
 const { getKnexClient } = require('@cumulus/db');
 const { unwrapDeadLetterCumulusMessage } = require('@cumulus/message/DeadLetterMessage');
 
 const { writeRecords } = require('./sf-event-sqs-to-db-records');
 
+/**
+ * Returns string of ISO Date Year/Month/Day
+ *
+ * @param {Date}   [date] - date
+ * @returns {string}
+ */
+const getISODate = (date = new Date()) => `${date.getFullYear()}-${date.getMonth()}-${date.getDate()}`;
+
+/**
+ * Generates new archive key for unprocessed dead letter message
+ *
+ * @param {string}   [failedKey] - key of message that failed to process
+ * @returns {string}
+ */
+const generateNewArchiveKeyForFailedMessage = (failedKey) => {
+  // Split key with format `${process.env.stackName}/dead-letter-archive/sqs/${messageId}`,
+  const splitKey = failedKey.split('/sqs/');
+  const pathPrefix = splitKey[0];
+  const messageId = splitKey[1];
+  return `${pathPrefix}/failed-sqs/${getISODate()}/${messageId}`;
+};
+
+/**
+ * Transfers unprocessed dead letters in the bucket to new location
+ * and deletes dead letters from old archive path
+ *
+ * @param {string}   [deadLetterMessage] - unprocessed dead letter message
+ * @param {string}   [bucket] - S3 bucket
+ * @returns {Promise<void>}
+ */
+const transferUnprocessedMessage = async (deadLetterMessage, bucket) => {
+  // Save allFailedKeys messages to different location
+  const s3KeyForFailedMessage = generateNewArchiveKeyForFailedMessage(deadLetterMessage.Key);
+  try {
```

Comment:
can we put both `await S3.s3PutObject` and `await deleteS3Object` in same try/catch?

Reply:
https://github.com/nasa/cumulus/pull/3099/commits/a738f3f1cf4a78d1454c73c6654d2d47938d341a#diff-bf4ec77eddad816778b1bfb695fa1e8f4efa13fe383ca4b595b36b8e234f6ffdR49-R55